wallet/src/Pos/Wallet/Web/Pending/Submission.hs

```diff
@@ -116,10 +118,22 @@ submitAndSavePtx PtxSubmissionHandlers{..} enqueue ptx@PendingTx{..} = do
     when walletTxCreationDisabled $
         throwM $ InternalError "Transaction creation is disabled by configuration!"
 
-    ack <- submitTxRaw enqueue _ptxTxAux
-    saveTx (_ptxTxId, _ptxTxAux) `catches` handlers ack
-    addOnlyNewPendingTx ptx
-    when ack $ ptxUpdateMeta _ptxWallet _ptxTxId PtxMarkAcknowledged
+    now <- getCurrentTimestamp
+    if | PtxApplying poolInfo <- _ptxCond,
+         Just creationTime <- poolInfo ^. thTimestamp,
+         diffTimestamp now creationTime > hour 1 -> do
+           let newCond = PtxWontApply "1h limit exceeded" poolInfo
+           void $ casPtxCondition _ptxWallet _ptxTxId _ptxCond newCond
+           logInfo $
+             sformat ("Pending transaction #"%build%" discarded becauce \
+                      \the 1h time limit was exceeded")
+                      _ptxTxId
+       | otherwise -> do
+           ack <- submitTxRaw enqueue _ptxTxAux
+           reportSubmitted ack
+           saveTx (_ptxTxId, _ptxTxAux) `catches` handlers ack
+           addOnlyNewPendingTx ptx
+           when ack $ ptxUpdateMeta _ptxWallet _ptxTxId PtxMarkAcknowledged
```

Comment:
Ok, so make sure we follow-up here with the fix for "save before submit" stuff, which is in a separate PR.

Reply:
Aye, this one, right @dcoutts ? https://github.com/input-output-hk/cardano-sl/pull/2270